source/Calamari.Shared/Extensions/VariableExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using Calamari.Commands.Support;
+using Calamari.Deployment;
+using Calamari.Integration.FileSystem;
+using Calamari.Util;
+
+namespace Calamari.Extensions
+{
+    public static class VariableExtensions
+    {
+        public static string GetPrimaryPackagePath(this IVariables variables, ICalamariFileSystem fileSystem, bool required)
```

Comment:
Not sure whether to keep this as an extension. It didn't seem appropriate on `CalamariVariables` and a bit too slim for a standalone service. Happy to take advice.

Reply:
In here seems ok for now. There's a lot going on in here, I'm wondering if a name like `GetPathToPrimaryPackage` might be a bit easier for a caller to understand. The current name just says to me that it's going to give me a path back, why it wants the fileSystem isn't clear unless I look at the implementation. I also wouldn't expect, based on the name alone, that it'd throw if the package didn't exist at that path. I might expect it's my job as the caller to check that once it's given me the path.

Reply:
I think the `required` parameter hints at that.